src/Repository/VolunteerRequestRepository.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\ApplicationRequest\VolunteerRequest;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\ORM\Query\Expr\Join;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+
+class VolunteerRequestRepository extends ServiceEntityRepository
+{
+    public function __construct(RegistryInterface $registry)
+    {
+        parent::__construct($registry, VolunteerRequest::class);
+    }
+
+    public function findForReferent(Adherent $referent): array
+    {
+        if (!$referent->isReferent()) {
+            return [];
+        }
+
+        return $this->createQueryBuilder('r')
+            ->select('r', 'CASE WHEN a.id IS NOT NULL THEN 1 ELSE 0 END AS isAdherent')
+            ->leftJoin(Adherent::class, 'a', Join::WITH, 'a.emailAddress = r.emailAddress AND a.adherent = 1')
```

Comment:
could you add the same check `AND a.adherent = 1` in `IsAdherentAdminExtension` please?